integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,391 @@
+        """Initialize the Watsonx client with configured credentials."""
+        credentials = Credentials(api_key=self.api_key.resolve_value(), url=self.api_base_url)
+
+        self.client = ModelInference(
+            model_id=self.model,
+            credentials=credentials,
+            project_id=self.project_id.resolve_value() if self.project_id else None,
+            verify=self.verify,
+            max_retries=self.max_retries,
+            delay_time=0.5,
+        )
+
+    def to_dict(self) -> dict[str, Any]:
+        """Serialize the component to a dictionary."""
+        return default_to_dict(
+            self,
+            model=self.model,
+            project_id=self.project_id.to_dict() if self.project_id else None,
+            api_base_url=self.api_base_url,
+            generation_kwargs=self.generation_kwargs,
+            api_key=self.api_key.to_dict(),
+            timeout=self.timeout,
+            max_retries=self.max_retries,
+            tools=self.tools,
+            verify=self.verify,
+        )
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "WatsonxChatGenerator":
+        """Deserialize the component from a dictionary."""
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key", "project_id"])
+        return default_from_dict(cls, data)
+
+    @component.output_types(replies=list[ChatMessage])
+    def run(
+        self,
+        messages: list[ChatMessage],
+        generation_kwargs: dict[str, Any] | None = None,
+        streaming_callback: StreamingCallbackT | None = None,
+    ) -> dict[str, list[ChatMessage]]:
+        """
+        Generate chat completions synchronously.
+
+        :param messages: List of ChatMessage objects
+        :param generation_kwargs: Additional generation parameters
+        :param streaming_callback: Callback function for streaming responses
+        :return: Dictionary with generated replies
+        """
+        if not messages:
+            return {"replies": []}
+
+        api_args = self._prepare_api_call(messages=messages, generation_kwargs=generation_kwargs)
+
+        if streaming_callback is not None:
+            return self._handle_streaming(api_args, streaming_callback)
+
+        return self._handle_standard(api_args)
+
+    @component.output_types(replies=list[ChatMessage])
+    async def run_async(
+        self,
+        messages: list[ChatMessage],
+        generation_kwargs: dict[str, Any] | None = None,
+        streaming_callback: StreamingCallbackT | None = None,
+    ) -> dict[str, list[ChatMessage]]:
+        """
+        Generate chat completions asynchronously.
+
+        :param messages: List of ChatMessage objects
+        :param generation_kwargs: Additional generation parameters
+        :param streaming_callback: Callback function for streaming responses
+        :return: Dictionary with generated replies
+        """
+        if not messages:
+            return {"replies": []}
+
```

Comment:
Same here add this code. Note that we set `requires_async=True` to enforce the streaming call back is async.
```suggestion
        streaming_callback = select_streaming_callback(
            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=True
        )
```